python/external_config/external_command.py

```diff
@@ -297,7 +297,7 @@ def tooltip(self):
         """
         return self._tooltip
 
-    def execute(self, pre_cache=False):
+    def execute(self, pre_cache=False, entity_ids=None):
```

Comment:
I am not 100% sure here, but perhaps we should just have a separate method called `execute_multiple_selection()` or `execute_shotgun_multiple_selection()` (is this only possible via the shotgun engine?) - to separate out this special case more from the normal use case. 

I don't think it's a legacy case - i  think we call it that in other places in the code too - at some point it was meant to become the legacy case but it turned out having multiple selection was an important aspect of some shotgun actions yet at the same time not applicable in other cases. 

The fact that this is so different and separate suggests that it might be worth exploring an interface with separate methods. But maybe that complicates the calling side of things?